Readme
Readme

This is a factorio mod called "Miles' BobsMods Expansion", currently build against factorio 0.16.51 and bobsmods (Whatever Version is newest and present during the 23rd December 2018)
The mod expands on a few machines such as assemblers and electronic assemblers but also adds a furnace tier between steel and electric: The electric steel furnace. Other features are still planned and can be submitted here on github


changelog.txt
-Initial release

-0.2.0 - 26th December 2018
Added the early electric furnace

-0.3.0 - 6th January 2019
Fixed technology naming issues not displaying correctly
Adjusted Science Time on technologies
Now with 100% more logos!

-0.4.0 - 12th January 2019
Fixed the early electric furnaces subgroup to be correct
Adjusted science time and cost to the early electric furnaces

-0.4.1 - 12th January 2019
Hotfix to add bob modules dependency for the items needed in the recipes
Hotfix on the item order for the early electric furnace

-0.5.0 - 26th July 2019
Updated for 0.17, sorry for the wait xoxo
Removed Bob's Plates dependency and updated dependency versions
Updated all Assembling Machines to HQ sprites
Changed Assembling Machine 8 to a new shade of pink (Thanks to LuziferSenpai!)

-0.5.1 - 27th July 2019
Hotfix for a dependency

-0.5.2 - 31st August 2019
Hotfix for a Typo, allowing 57 Modules in an Assembler instead of the desired 7

-0.6.0 - 19st November 2019
New Icon Art
Nerfed Crafting Speed of normal Assemblers to a maximum of 5.00
Updated Dependencies to their newest versions

-0.6.1 - 14th December 2019
Fixed Pollution of electronic Assemblers, T4 generating 0.1 less thant T3 and T5 generating 0.1 less than T4
Adjusted Module Slots to 7 and 8 for electronic assembler T4 and T5 (Down from 8 and 10)
Adjusted energy usage of electronic assembler T4 and T5 to 800kW and 1000kW to be more in line with the other machines

-0.7.0 - 6th March 2020
Updated for 0.18, again: Was being lazy, literally just had to change version numbers xoxo

-0.8.0 - 6th July 2020
Happy Birthday to me! Spent the night adding conditional recipes. Depending if bobplates and/or bobelectronics are present, recipes change
Adjusted crafting speed and module slots
Fixed technology icon as well as some machine icons

-0.8.3 - 20th July 2020
Hotfix for crafting speed of assemblers. T7 now actually better than T6. Raised maximum craft speed of all machines
Adjusted conditional recipes for Bob's Plates and/or Bobs'Electronics to be more in-line with high tier metals. Additional CPU requirement removed.

prototypes/assembly.lua
require "util"
data:extend(
{
  {
    type = "item",
    name = "assembling-machine-7",
    icon = "__MilesBobsExpansion__/graphics/icons/assembling-machine-7.png",
    icon_size = 32,
    subgroup = "bob-assembly-machine",
    order = "c[assembling-machine-7]",
    place_result = "assembling-machine-7",
    stack_size = 50
  },

  {
    type = "assembling-machine",
    name = "assembling-machine-7",
    icon = "__MilesBobsExpansion__/graphics/icons/assembling-machine-7.png",
    icon_size = 32,
    flags = {"placeable-neutral","placeable-player", "player-creation"},
    minable = {hardness = 0.2, mining_time = 0.5, result = "assembling-machine-7"},
    max_health = 800,
    corpse = "big-remnants",
    dying_explosion = "medium-explosion",
    resistances = 
    {
      {
        type = "fire",
        percent = 70
      }
    },
    fluid_boxes =
    {
      {
        production_type = "input",
        pipe_picture = assembler2pipepictures(),
        pipe_covers = pipecoverspictures(),
        base_area = 10,
        base_level = -1,
        pipe_connections = {{ type="input", position = {0, -2} }}
      },
      {
        production_type = "output",
        pipe_picture = assembler2pipepictures(),
        pipe_covers = pipecoverspictures(),
        base_area = 10,
        base_level = 1,
        pipe_connections = {{ type="output", position = {0, 2} }}
      },
      off_when_no_fluid_recipe = true
    },
    collision_box = {{-1.2, -1.2}, {1.2, 1.2}},
    selection_box = {{-1.5, -1.5}, {1.5, 1.5}},
    fast_replaceable_group = "assembling-machine",
    next_upgrade = "assembling-machine-8",
    animation =
    {
      layers =
      {
        {
          filename = "__bobassembly__/graphics/entity/assembling-machine/assembling-machine-3.png",
          priority = "high",
          width = 108,
          height = 119,
          frame_count = 32,
          line_length = 8,
          shift = util.by_pixel(0, -0.5),
          hr_version =
          {
            filename = "__bobassembly__/graphics/entity/assembling-machine/hr-assembling-machine-3.png",
            priority = "high",
            width = 214,
            height = 237,
            frame_count = 32,
            line_length = 8,
            shift = util.by_pixel(0, -0.75),
            scale = 0.5
          }
        },
        {
          filename = "__bobassembly__/graphics/entity/assembling-machine/assembling-machine-mask.png",
          priority = "high",
          width = 142,
          height = 113,
          repeat_count = 32,
          shift = {0.84, -0.09},
          tint = {r = 0.4, g = 0.6, b = 0.6},
        },
        {
          filename = "__bobassembly__/graphics/entity/assembling-machine/assembling-machine-3-shadow.png",
          priority = "high",
          width = 130,
          height = 82,
          frame_count = 32,
          line_length = 8,
          draw_as_shadow = true,
          shift = util.by_pixel(28, 4),
          hr_version =
          {
            filename = "__bobassembly__/graphics/entity/assembling-machine/hr-assembling-machine-3-shadow.png",
            priority = "high",
            width = 260,
            height = 162,
            frame_count = 32,
            line_length = 8,
            draw_as_shadow = true,
            shift = util.by_pixel(28, 4),
            scale = 0.5
          }
        }
      }
    },
    open_sound = { filename = "__base__/sound/machine-open.ogg", volume = 0.85 },
    close_sound = { filename = "__base__/sound/machine-close.ogg", volume = 0.75 },
    working_sound =
    {
      sound = {
        {
          filename = "__base__/sound/assembling-machine-t2-1.ogg",
          volume = 0.8
        },
        {
          filename = "__base__/sound/assembling-machine-t2-2.ogg",
          volume = 0.8
        },
      },
      idle_sound = { filename = "__base__/sound/idle1.ogg", volume = 0.6 },
      apparent_volume = 1.5,
    },
    crafting_categories = {"crafting", "advanced-crafting", "crafting-with-fluid"},
    crafting_speed = 4,
    energy_source =
    {
      type = "electric",
      usage_priority = "secondary-input",
      emissions_per_minute = 0.2
    },
    energy_usage = "650kW",
    module_specification =
    {
      module_slots = 6,
      module_info_icon_shift = {0, 0.5},
      module_info_multi_row_initial_height_modifier = -0.3
    },
    allowed_effects = {"consumption", "speed", "productivity", "pollution"}
  },

  {
    type = "recipe",
    name = "assembling-machine-7",
    enabled = "false",
    ingredients =
    {
      {"assembling-machine-6", 1},
      {"processing-unit", 10},
      {"steel-plate", 10},
      {"iron-plate", 15},
      {"iron-gear-wheel", 10},
	  {"speed-module-5", 5},
    },
    result = "assembling-machine-7"
  },

  {
    type = "technology",
    name = "automation-7",
    icon = "__base__/graphics/technology/automation.png",
    icon_size = 128,
    effects =
    {
      {
        type = "unlock-recipe",
        recipe = "assembling-machine-7"
      }
    },
    prerequisites = {"automation-6"},
    unit =
    {
      count = 200,
      ingredients =
      {
        {"automation-science-pack", 1},
        {"logistic-science-pack", 1},
        {"utility-science-pack", 1},
        {"production-science-pack", 1},
      },
      time = 100
    },
    upgrade = true,
  },

  {
    type = "recipe",
    name = "assembling-machine-8",
    enabled = "false",
    ingredients =
    {
      {"assembling-machine-7", 1},
      {"processing-unit", 15},
      {"steel-plate", 15},
      {"iron-plate", 20},
      {"iron-gear-wheel", 15},
	  {"speed-module-6", 5},
    },
    result = "assembling-machine-8"
  },

  {
    type = "technology",
    name = "automation-8",
    icon = "__base__/graphics/technology/automation.png",
    icon_size = 128,
    effects =
    {
      {
        type = "unlock-recipe",
        recipe = "assembling-machine-8"
      }
    },
    prerequisites = {"automation-7"},
    unit =
    {
      count = 250,
      ingredients =
      {
        {"automation-science-pack", 1},
        {"logistic-science-pack", 1},
        {"utility-science-pack", 1},
        {"production-science-pack", 1},
      },
      time = 150
	  },
	upgrade = true,
  },

  {
    type = "recipe",
    name = "assembling-machine-9",
    enabled = "false",
    ingredients =
    {
      {"assembling-machine-8", 1},
      {"processing-unit", 20},
      {"steel-plate", 20},
      {"iron-plate", 40},
      {"iron-gear-wheel", 30},
	  {"speed-module-7", 5},
    },
    result = "assembling-machine-9"
  },

  {
    type = "technology",
    name = "automation-9",
    icon = "__base__/graphics/technology/automation.png",
    icon_size = 128,
    effects =
    {
      {
        type = "unlock-recipe",
        recipe = "assembling-machine-9"
      }
    },
    prerequisites = {"automation-8"},
    unit =
    {
      count = 450,
      ingredients =
      {
        {"automation-science-pack", 1},
        {"logistic-science-pack", 1},
        {"utility-science-pack", 1},
        {"production-science-pack", 1},
      },
      time = 200
	  },
	upgrade = true,
  },

  {
    type = "recipe",
    name = "assembling-machine-10",
    enabled = "false",
    ingredients =
    {
      {"assembling-machine-9", 1},
      {"processing-unit", 30},
      {"steel-plate", 40},
      {"iron-plate", 80},
      {"iron-gear-wheel", 60},
	  {"speed-module-8", 5},
    },
    result = "assembling-machine-10"
  },

  {
    type = "technology",
    name = "automation-10",
    icon = "__base__/graphics/technology/automation.png",
    icon_size = 128,
    effects =
    {
      {
        type = "unlock-recipe",
        recipe = "assembling-machine-10"
      }
    },
    prerequisites = {"automation-9"},
    unit =
    {
      count = 600,
      ingredients =
      {
        {"automation-science-pack", 1},
        {"logistic-science-pack", 1},
        {"utility-science-pack", 1},
        {"production-science-pack", 1},
      },
      time = 250
	  },
	upgrade = true,
  },
}
)
		local new_assembler=table.deepcopy(data.raw["assembling-machine"]["assembling-machine-7"])
		new_assembler.name="assembling-machine-8"
		new_assembler.icon="__MilesBobsExpansion__/graphics/icons/assembling-machine-8.png"
		new_assembler.crafting_speed="5.0"
		new_assembler.order="c[assembling-machine-8]"
		new_assembler.animation.layers[2].tint ={r = 243/255, g = 136/255, b = 213/255}
		new_assembler.energy_usage="1000kW"
		new_assembler.minable={hardness = 0.2, mining_time = 0.5, result = "assembling-machine-8"}
		new_assembler.module_specification.module_slots="6"
		data:extend{new_assembler}
		
		local new_assembleritem=table.deepcopy(data.raw["item"]["assembling-machine-7"])
        new_assembleritem.name="assembling-machine-8"
		new_assembleritem.place_result="assembling-machine-8"
        new_assembler.order="c[assembling-machine-8]"		
		new_assembleritem.icon="__MilesBobsExpansion__/graphics/icons/assembling-machine-8.png"
		data:extend{new_assembleritem}
		
		local new_assembler=table.deepcopy(data.raw["assembling-machine"]["assembling-machine-7"])
		new_assembler.name="assembling-machine-9"
		new_assembler.icon="__MilesBobsExpansion__/graphics/icons/assembling-machine-9.png"
		new_assembler.crafting_speed="5.50"
		new_assembler.order="c[assembling-machine-9]"
		new_assembler.animation.layers[2].tint={r = 0.9, g = 0.7, b = 0.0}
		new_assembler.energy_usage="1250kW"
		new_assembler.minable={hardness = 0.2, mining_time = 0.5, result = "assembling-machine-9"}
		new_assembler.module_specification.module_slots="7"
		data:extend{new_assembler}
		
		local new_assembleritem=table.deepcopy(data.raw["item"]["assembling-machine-7"])
        new_assembleritem.name="assembling-machine-9"
		new_assembleritem.place_result="assembling-machine-9"
		new_assembler.order="c[assembling-machine-9]"
		new_assembleritem.icon="__MilesBobsExpansion__/graphics/icons/assembling-machine-9.png"
		data:extend{new_assembleritem}
		
		local new_assembler=table.deepcopy(data.raw["assembling-machine"]["assembling-machine-7"])
		new_assembler.name="assembling-machine-10"
		new_assembler.icon="__MilesBobsExpansion__/graphics/icons/assembling-machine-10.png"
		new_assembler.crafting_speed="6.00"
		new_assembler.order="c[x-assembling-machine-9+1]"
		new_assembler.animation.layers[2].tint={r = 1.0, g = 1.0, b = 1.0}
		new_assembler.energy_usage="2000kW"
		new_assembler.minable={hardness = 0.2, mining_time = 0.5, result = "assembling-machine-10"}
		new_assembler.module_specification.module_slots="8"
		data:extend{new_assembler}
		
		local new_assembleritem=table.deepcopy(data.raw["item"]["assembling-machine-7"])
        new_assembleritem.name="assembling-machine-10"
		new_assembleritem.place_result="assembling-machine-10"
    	new_assembleritem.order="c[assembling-machine-9+1]"
		new_assembleritem.icon="__MilesBobsExpansion__/graphics/icons/assembling-machine-10.png"
		data:extend{new_assembleritem}
		
		
		local recipe = {
			name = "assembling-machine-7",
			type = "recipe",
			ingredients =
				{
					{"assembling-machine-6", 1},
					{"processing-unit", 5},
					{"steel-plate", 10},
					{"iron-plate", 10},
					{"iron-gear-wheel", 10},
					{"speed-module-5", 5},
				},
			result = "assembling-machine-7"
		}
		if mods["bobplates"] then
			recipe.ingredients =
				{
					{"assembling-machine-6", 1},
					{"processing-unit", 5},
					{"tungsten-carbide", 10},
					{"tungsten-gear-wheel", 10},
					{"speed-module-5", 5},
				}
		end
		
		if mods["bobelectronics"] then
			recipe.ingredients =
				{
					{"assembling-machine-6", 1},
					{"advanced-processing-unit", 5},
					{"steel-plate", 10},
					{"iron-plate", 10},
					{"iron-gear-wheel", 10},
					{"speed-module-5", 5},
				}
		end
		
		if mods["bobplates"] and mods["bobelectronics"] then
			recipe.ingredients =
				{
					{"assembling-machine-6", 1},
					{"advanced-processing-unit", 5},
					{"tungsten-carbide", 10},
					{"tungsten-gear-wheel", 10},
					{"speed-module-5", 5},
				}
		end
		data:extend({recipe})
		
		local recipe = {
			name = "assembling-machine-8",
			type = "recipe",
			ingredients =
				{
					{"assembling-machine-7", 1},
					{"processing-unit", 8},
					{"steel-plate", 15},
					{"iron-plate", 15},
					{"iron-gear-wheel", 15},
					{"speed-module-6", 5},
				},
			result = "assembling-machine-8"
		}
		if mods["bobplates"] then
			recipe.ingredients =
				{
					{"assembling-machine-7", 1},
					{"processing-unit", 8},
					{"tungsten-carbide", 10},
					{"nitinol-alloy", 20},
					{"speed-module-6", 5},
				}
		end
		
		if mods["bobelectronics"] then
			recipe.ingredients =
				{
					{"assembling-machine-7", 1},
					{"advanced-processing-unit", 8},
					{"steel-plate", 15},
					{"iron-plate", 15},
					{"iron-gear-wheel", 15},
					{"speed-module-6", 5},
				}
		end
		
		if mods["bobplates"] and mods["bobelectronics"] then
			recipe.ingredients =
				{
					{"assembling-machine-7", 1},
					{"advanced-processing-unit", 8},
					{"tungsten-carbide", 10},
					{"nitinol-alloy", 20},
					{"speed-module-6", 5},
				}
		end
		data:extend({recipe})
				
		local recipe = {
			name = "assembling-machine-9",
			type = "recipe",
			ingredients =
				{
					{"assembling-machine-8", 1},
					{"processing-unit", 10},
					{"steel-plate", 20},
					{"iron-plate", 20},
					{"iron-gear-wheel", 20},
					{"speed-module-7", 5},
    },
			result = "assembling-machine-9"
		}
		if mods["bobplates"] then
			recipe.ingredients =
				{
					{"assembling-machine-8", 1},
					{"processing-unit", 10},
					{"copper-tungsten-alloy", 10},
					{"tungsten-carbide", 20},
					{"nitinol-bearing", 20},
					{"speed-module-7", 5},
				}
		end
		
		if mods["bobelectronics"] then
			recipe.ingredients =
				{
					{"assembling-machine-8", 1},
					{"advanced-processing-unit", 10},
					{"steel-plate", 20},
					{"iron-plate", 20},
					{"iron-gear-wheel", 20},
					{"speed-module-7", 5},
				}
		end
		
		if mods["bobplates"] and mods["bobelectronics"] then
			recipe.ingredients =
				{
					{"assembling-machine-8", 1},
					{"advanced-processing-unit", 10},
					{"copper-tungsten-alloy", 10},
					{"tungsten-carbide", 20},
					{"nitinol-bearing", 20},
					{"speed-module-7", 5},
				}
		end
		data:extend({recipe})
				
		local recipe = {
			name = "assembling-machine-10",
			type = "recipe",
			ingredients =
				{
					{"assembling-machine-9", 1},
					{"processing-unit", 20},
					{"steel-plate", 30},
					{"iron-plate", 30},
					{"iron-gear-wheel", 30},
					{"speed-module-8", 5},
				},
			result = "assembling-machine-10"
		}
		if mods["bobplates"] then
			recipe.ingredients =
				{
					{"assembling-machine-9", 1},
					{"processing-unit", 20},
					{"copper-tungsten-alloy", 20},
					{"tungsten-carbide", 20},
					{"nitinol-bearing", 20},
					{"speed-module-8", 5},
				}
		end
		
		if mods["bobelectronics"] then
			recipe.ingredients =
				{
					{"assembling-machine-9", 1},
					{"advanced-processing-unit", 20},
					{"steel-plate", 30},
					{"iron-plate", 30},
					{"iron-gear-wheel", 30},
					{"speed-module-8", 5},
				}
		end
		
		if mods["bobplates"] and mods["bobelectronics"] then
			recipe.ingredients =
				{
					{"assembling-machine-9", 1},
					{"advanced-processing-unit", 20},
					{"copper-tungsten-alloy", 20},
					{"nitinol-alloy", 20},
					{"nitinol-bearing", 20},
					{"speed-module-8", 5},
				}
		end
		data:extend({recipe})

prototypes/assembly-electronics.lua
require "util"
data:extend(
{
  {
    type = "item",
    name = "electronics-machine-4",
    icon = "__MilesBobsExpansion__/graphics/icons/electronics-machine-4.png",
    icon_size = 32,
    subgroup = "bob-assembly-machine",
    order = "d[electronics-machine-4]",
    place_result = "electronics-machine-4",
    stack_size = 50
  },

  {
    type = "assembling-machine",
    name = "electronics-machine-4",
    icon = "__MilesBobsExpansion__/graphics/icons/electronics-machine-4.png",
    icon_size = 32,
    flags = {"placeable-neutral", "placeable-player", "player-creation"},
    minable = {hardness = 0.2, mining_time = 0.5, result = "electronics-machine-4"},
    max_health = 600,
    corpse = "medium-remnants",
    dying_explosion = "medium-explosion",
    resistances =
    {
      {
        type = "fire",
        percent = 70
      }
    },
    fluid_boxes =
    {
      {
        production_type = "input",
        pipe_picture = assembler3pipepictures(),
        pipe_covers = pipecoverspictures(),
        base_area = 10,
        base_level = -1,
        pipe_connections = {{ type="input", position = {0.5, -1.5} }}
      },
      off_when_no_fluid_recipe = true
    },
    collision_box = {{-0.7, -0.7}, {0.7, 0.7}},
    selection_box = {{-1, -1}, {1, 1}},
    fast_replaceable_group = "assembling-machine",
	next_upgrade = "electronics-machine-5",
	animation =
    {
      layers =
      {
        {
          filename = "__bobassembly__/graphics/entity/assembling-machine/assembling-machine-2.png",
          priority = "high",
          width = 108,
          height = 110,
          frame_count = 32,
          line_length = 8,
          shift = util.by_pixel(0, 4 *2/3),
          scale = 2/3,
          hr_version =
          {
            filename = "__bobassembly__/graphics/entity/assembling-machine/hr-assembling-machine-2.png",
            priority = "high",
            width = 214,
            height = 218,
            frame_count = 32,
            line_length = 8,
            shift = util.by_pixel(0, 4 *2/3),
            scale = 1/3
          }
        },
        {
          filename = "__bobassembly__/graphics/entity/assembling-machine/assembling-machine-mask.png",
          priority = "high",
          width = 142,
          height = 113,
          repeat_count = 32,
          shift = {0.84 *2/3, -0.09 *2/3},
          scale = 2/3,
          tint = {r = 0.7, g = 0.2, b = 0.1},
        },
        {
          filename = "__bobassembly__/graphics/entity/assembling-machine/assembling-machine-2-shadow.png",
          priority = "high",
          width = 98,
          height = 82,
          frame_count = 32,
          line_length = 8,
          draw_as_shadow = true,
          shift = util.by_pixel(12 *2/3, 5 *2/3),
          scale = 2/3,
          hr_version =
          {
            filename = "__bobassembly__/graphics/entity/assembling-machine/hr-assembling-machine-2-shadow.png",
            priority = "high",
            width = 196,
            height = 163,
            frame_count = 32,
            line_length = 8,
            draw_as_shadow = true,
            shift = util.by_pixel(12 *2/3, 4.75 *2/3),
            scale = 1/3
          }
        }
      }
    },
    open_sound = { filename = "__base__/sound/machine-open.ogg", volume = 0.85 },
    close_sound = { filename = "__base__/sound/machine-close.ogg", volume = 0.75 },
    working_sound =
    {
      sound = {
        {
          filename = "__base__/sound/assembling-machine-t3-1.ogg",
          volume = 0.8
        },
        {
          filename = "__base__/sound/assembling-machine-t3-2.ogg",
          volume = 0.8
        },
      },
      idle_sound = { filename = "__base__/sound/idle1.ogg", volume = 0.6 },
      apparent_volume = 1.5,
    },
    crafting_categories = {"electronics", "electronics-machine"},
    crafting_speed = 6,
    energy_source =
    {
      type = "electric",
      usage_priority = "secondary-input",
      emissions = 0.0005
    },
    energy_usage = "800kW",
    ingredient_count = 6,
    module_specification =
    {
      module_slots = 6,
    },
    allowed_effects = {"consumption", "speed", "productivity", "pollution"}
  },


  {
    type = "recipe",
    name = "electronics-machine-4",
    enabled = "false",
    ingredients =
    {
      {"electronics-machine-3", 1},
      {"processing-unit", 10},
      {"steel-plate", 10},
      {"iron-gear-wheel", 10},
	  {"speed-module-3", 5},
    },
    result = "electronics-machine-4"
  },

  {
    type = "technology",
    name = "electronics-machine-4",
    icon = "__base__/graphics/technology/automation.png",
    icon_size = 128,
    effects =
    {
      {
        type = "unlock-recipe",
        recipe = "electronics-machine-4"
      }
    },
    prerequisites =
    {
      "electronics-machine-3",
      "advanced-electronics-2"
    },
    unit =
    {
      count = 200,
      time = 150,
      ingredients =
      {
        {"automation-science-pack", 1},
        {"logistic-science-pack", 1},
        {"utility-science-pack", 1},
        {"production-science-pack", 1},
      },
    },
    upgrade = true,
  },
    {
    type = "recipe",
    name = "electronics-machine-5",
    enabled = "false",
    ingredients =
    {
      {"electronics-machine-4", 1},
      {"processing-unit", 20},
      {"steel-plate", 20},
      {"iron-gear-wheel", 20},
	  {"speed-module-4", 5},
    },
    result = "electronics-machine-5"
  },

  {
    type = "technology",
    name = "electronics-machine-5",
    icon = "__base__/graphics/technology/automation.png",
    icon_size = 128,
    effects =
    {
      {
        type = "unlock-recipe",
        recipe = "electronics-machine-5"
      }
    },
    prerequisites =
    {
      "electronics-machine-4",
      "advanced-electronics-2"
    },
    unit =
    {
      count = 400,
      time = 200,
      ingredients =
      {
        {"automation-science-pack", 1},
        {"logistic-science-pack", 1},
        {"utility-science-pack", 1},
        {"production-science-pack", 1},
      },
    },
    upgrade = true,
  },
}
)
		local new_assembler=table.deepcopy(data.raw["assembling-machine"]["electronics-machine-4"])
		new_assembler.name="electronics-machine-5"
		new_assembler.icon="__MilesBobsExpansion__/graphics/icons/electronics-machine-5.png"
		new_assembler.crafting_speed="8.0"
		new_assembler.order="d[electronics-machine-5]"
		new_assembler.animation.layers[2].tint={r = 1.0, g = 1.0, b = 1.0}
		new_assembler.energy_usage="1000kW"
		new_assembler.energy_source={type = "electric", usage_priority = "secondary-input", emissions = 0.0003}
		new_assembler.minable={hardness = 0.2, mining_time = 0.5, result = "electronics-machine-5"}
		new_assembler.module_specification.module_slots="6"
		data:extend{new_assembler}
		
		local new_assembleritem=table.deepcopy(data.raw["item"]["electronics-machine-4"])
        new_assembleritem.name="electronics-machine-5"
		new_assembleritem.place_result="electronics-machine-5"		
		new_assembleritem.icon="__MilesBobsExpansion__/graphics/icons/electronics-machine-5.png"
		data:extend{new_assembleritem}
		
				
		local recipe = {
			name = "electronics-machine-4",
			type = "recipe",
			ingredients =
				{
					{"electronics-machine-3", 1},
					{"processing-unit", 10},
					{"steel-plate", 10},
					{"iron-gear-wheel", 10},
					{"speed-module-3", 5},
				},
			result = "electronics-machine-4"
		}
		if mods["bobplates"] then
			recipe.ingredients =
				{
					{"electronics-machine-3", 1},
					{"processing-unit", 10},
					{"nitinol-alloy", 10},
					{"nitinol-bearing", 10},
					{"tungsten-carbide", 10},
					{"speed-module-5", 5},
				}
		end
		
		if mods["bobelectronics"] then
			recipe.ingredients =
				{
					{"electronics-machine-3", 1},
					{"advanced-processing-unit", 5},
					{"steel-plate", 10},
					{"iron-gear-wheel", 10},
					{"speed-module-5", 5},
				}
		end
		
		if mods["bobplates"] and mods["bobelectronics"] then
			recipe.ingredients =
				{
					{"electronics-machine-3", 1},
					{"advanced-processing-unit", 10},
					{"nitinol-alloy", 10},
					{"nitinol-bearing", 10},
					{"tungsten-carbide", 10},
					{"speed-module-5", 5},
				}
		end
		data:extend({recipe})	

		
		local recipe = {
			name = "electronics-machine-5",
			type = "recipe",
			ingredients =
				{
					{"electronics-machine-4", 1},
					{"processing-unit", 20},
					{"steel-plate", 20},
					{"iron-gear-wheel", 20},
					{"speed-module-3", 5},
				},
			result = "electronics-machine-5"
		}
		if mods["bobplates"] then
			recipe.ingredients =
				{
					{"electronics-machine-4", 1},
					{"processing-unit", 20},
					{"copper-tungsten-alloy", 20},
					{"nitinol-bearing", 20},
					{"speed-module-6", 5},
				}
		end
		
		if mods["bobelectronics"] then
			recipe.ingredients =
				{
					{"electronics-machine-4", 1},
					{"advanced-processing-unit", 20},
					{"steel-plate", 20},
					{"iron-gear-wheel", 20},
					{"speed-module-6", 5},
				}
		end
		
		if mods["bobplates"] and mods["bobelectronics"] then
			recipe.ingredients =
				{
					{"electronics-machine-4", 1},
					{"advanced-processing-unit", 20},
					{"copper-tungsten-alloy", 20},
					{"nitinol-bearing", 20},
					{"speed-module-6", 5},
				}
		end
		data:extend({recipe})